mobile/src/screens/marketplace.js

```diff
@@ -584,6 +626,7 @@ class MarketplaceScreen extends Component {
         onLoadEnd={this.onLoadEnd}
         onError={this.onError}
         onNavigationStateChange={this.onNavigationStateChange}
+        onShouldStartLoadWithRequest={this.onNavigationStateChange}
```

Comment:
I think we can probably use onShouldStartLoadWithRequest everywhere here and not have to use the `this.goBackToDapp()` call. Do you think that'd work @shahthepro? It would be a lot cleaner.

Reply:
@tomlinton Yeah, I did try that one. But as I remember, it worked only on one platform (Android). It didn't work on iOS.

Reply:
Does that mean we are using two different paths here, one for iOS and one for Android? If possible we should just use the same for both (i.e. if `onShouldStartLoadWithRequest` doesn't work on iOS we probably should just use `onNavigationStateChange`).

Reply:
Okay, going through react web view docs again, [`onShouldStartLoadWithRequest`](https://facebook.github.io/react-native/docs/webview#onshouldstartloadwithrequest) works in iOS but not on Android. Sorry, a blunder from my side.

For Android, we use the [`stopLoading()`](https://facebook.github.io/react-native/docs/webview#stoploading) method to prevent the page from continuing further. 

https://github.com/OriginProtocol/origin/pull/3454/files#diff-c0d3f54b8a00383ba9f0b8ce455b87b4R519

`stopLoading()` works both on iOS and Android. But on iOS, `onShouldStartLoadWithRequest` seemed like a cleaner way to do it.

Can change this though, if necessary.

Reply:
The WebView package has been extracted from `react-native` and has now diverged (the WebView component will be removed from `react-native` core). We use the one here [which is a lot more up to date](https://github.com/react-native-community/react-native-webview). I believe it should work for both iOS and Android. See https://github.com/react-native-community/react-native-webview/blob/master/docs/Reference.md#onshouldstartloadwithrequest

Reply:
I played around with `onShouldStartLoadWithRequest` but not sure if we can get it to work as it appears to not support async functions.